backend-project/small_eod/generic/tests/test_views.py

```diff
@@ -3,7 +3,38 @@
 from ...users.mixins import AuthenticatedMixin
 
 
-class ReadOnlyViewSetMixin(AuthenticatedMixin):
+class NumQueriesLimitMixin:
+    def test_num_queries_for_list(self):
+        self.login_required()
+        if not hasattr(self, "get_url_list"):
+            raise NotImplementedError(
+                "NumQueriesLimit mixin must be used alongside the ReadOnlyViewSetMixin"
+            )
+        existing_instances_num = self.factory_class._meta.model.objects.all().count()
+        with self.assertNumQueriesLessThan(self.queries_less_than_limit):
+            response = self.client.get(self.get_url_list())
+        self.assertEqual(response.status_code, 200)
+        assert response.json()["count"] == existing_instances_num
+
+        # number of queries after adding a new instance
+        self.new_obj = self.factory_class()
```

Comment:
```suggestion
        self.factory_class.create_batch(size=5)
```
W ten sposób szybciej wyłapiemy problemy.

Czy to przypisanie do ```self.``` jest konieczne?